addons/html_builder/static/src/snippets/snippet_service.js

```diff
@@ -274,6 +274,18 @@ export class SnippetModel extends Reactive {
             return;
         }
         snippet.title = newName;
+        for (const snippetEl of this.snippetsDocument.body.querySelectorAll("snippets#snippet_custom > *")) {
+            if (snippetEl.getAttribute("data-oe-snippet-key") === snippet.key) {
+                snippetEl.setAttribute("name", newName);
+                snippetEl.children[0].dataset["name"] = newName;
+            }
+        }
+
+        const snippetDoc = document.querySelector("iframe.o_add_snippet_iframe").contentDocument;
+        const snippetSection = snippetDoc.querySelector(`[data-snippet-id="${snippet.key}"] > section`);
+        if (snippetSection) {
+            snippetSection.dataset["name"] = newName;
+        }
```

Comment:
Why not edit snippet.content ?
snippet.content.dataset["name"] = newName 

It s not enought ?